src/concurrency.md

```diff
@@ -33,6 +34,40 @@ fn main() {
 }
 ```
 
+[ex-rayon-any-all]: #ex-rayon-any-all
+<a name="ex-rayon-any-all"></a>
+## Check the elements of an array against a predicate in parallel
+
+[![rayon-badge]][rayon] [![cat-concurrency-badge]][cat-concurrency]
+
+The `rayon` provides a [`ParallelIterator`] trait which includes parallelized implementations of many of the methods from the standard library's [`Iterator`] trait.
+
+This example demonstrates using the [`any`] and [`all`] methods. [`any`] checks in parallel whether any element of the iterator matches the predicate, and returns as soon as one is found. [`all`] checks in parallel whether all elements of the iterator match the predicate, and returns as soon as a non-matching element is found.
```

Comment:
I think we should use `rayon::any` and `rayon::all` here, since there are also functions `any` and `all` in the standard library. The concern isn't so much clarity (I think this example is very well written) but more so about namespacing the function names in our links.